zebra-rpc/src/server/rpc_call_compatibility.rs

```diff
@@ -3,116 +3,66 @@
-    fn call_description(call: &Call) -> String {
-        const MAX_PARAMS_LOG_LENGTH: usize = 100;
+impl<'a> RpcServiceT<'a> for FixRpcResponseMiddleware {
+    type Future = ResponseFuture<futures::future::BoxFuture<'a, jsonrpsee::MethodResponse>>;
 
-        match call {
-            Call::MethodCall(MethodCall { method, params, .. }) => {
-                let mut params = format!("{params:?}");
-                if params.len() >= MAX_PARAMS_LOG_LENGTH {
-                    params.truncate(MAX_PARAMS_LOG_LENGTH);
-                    params.push_str("...");
-                }
+    fn call(&self, request: jsonrpsee::types::Request<'a>) -> Self::Future {
+        let service = self.service.clone();
+        ResponseFuture::future(Box::pin(async move {
+            let response = service.call(request).await;
+            if response.is_error() {
+                let original_error_code = response
+                    .as_error_code()
+                    .expect("response should have an error code");
+                if original_error_code == jsonrpsee_types::ErrorCode::InvalidParams.code() {
+                    let new_error_code = crate::server::error::LegacyCode::Misc.into();
+                    tracing::debug!(
+                        "Replacing RPC error: {original_error_code} with {new_error_code}"
+                    );
+                    let json: serde_json::Value =
+                        serde_json::from_str(response.into_parts().0.as_str())
+                            .expect("response string should be valid json");
+                    let id = json["id"]
+                        .as_str()
+                        .expect("response json should have an id")
+                        .to_string();
 
-                format!(r#"method = {method:?}, params = {params}"#)
-            }
-            Call::Notification(Notification { method, params, .. }) => {
-                let mut params = format!("{params:?}");
-                if params.len() >= MAX_PARAMS_LOG_LENGTH {
-                    params.truncate(MAX_PARAMS_LOG_LENGTH);
-                    params.push_str("...");
+                    return MethodResponse::error(
+                        jsonrpsee_types::Id::Str(id.into()),
+                        ErrorObject::borrowed(new_error_code, "Invalid params", None),
+                    );
                 }
-
-                format!(r#"notification = {method:?}, params = {params}"#)
             }
-            Call::Invalid { .. } => "invalid request".to_owned(),
-        }
-    }
-
-    /// Check RPC output and log any errors.
-    //
-    // TODO: do we want to ignore ErrorCode::ServerError(_), or log it at debug?
-    fn log_if_error(output: &Option<Output>, call: Call) {
-        if let Some(Output::Failure(Failure { error, .. })) = output {
-            let call_description = Self::call_description(&call);
-            tracing::info!("RPC error: {error} in call: {call_description}");
```

Comment:
Logging errors here would be nice if it's still possible.

Reply:
I am not totally sure if this is a good idea as the log can be triggered by the caller. We have the cookie auth method that should legitimate the end user but it can be disabled. I think we can leave this one out and add it in a future PR if we decide to do it.